components/dash-core-components/src/fragments/Dropdown.react.js

```diff
@@ -21,90 +21,99 @@ const TOKENIZER = {
-                    (loading_state && loading_state.is_loading) || undefined
+    const onInputChange = useCallback(
+        search_value => setProps({search_value}),
+        []
+    );
+
+    useEffect(() => {
+        if (optionsCheck !== sanitizedOptions && !isNil(value)) {
+            const values = sanitizedOptions.map(option => option.value);
+            if (multi && Array.isArray(value)) {
+                const invalids = value.filter(v => !values.includes(v));
+                if (invalids.length) {
+                    setProps({value: without(invalids, value)});
                 }
-            >
-                <ReactDropdown
-                    filterOptions={filterOptions}
-                    options={sanitizeOptions(options)}
-                    value={selectedValue}
-                    onChange={selectedOption => {
-                        if (multi) {
-                            let value;
-                            if (isNil(selectedOption)) {
-                                value = [];
-                            } else {
-                                value = pluck('value', selectedOption);
-                            }
-                            setProps({value});
-                        } else {
-                            let value;
-                            if (isNil(selectedOption)) {
-                                value = null;
-                            } else {
-                                value = selectedOption.value;
-                            }
-                            setProps({value});
-                        }
-                    }}
-                    onInputChange={search_value => setProps({search_value})}
-                    backspaceRemoves={clearable}
-                    deleteRemoves={clearable}
-                    inputProps={{autoComplete: 'off'}}
-                    {...omit(['setProps', 'value', 'options'], this.props)}
-                />
-            </div>
-        );
-    }
-}
+            } else {
+                if (!values.includes(value)) {
+                    setProps({value: null});
+                }
+            }
+            setOptionsCheck(sanitizedOptions);
+        }
+    }, [sanitizedOptions, optionsCheck, multi, value]);
+
+    return (
+        <div
+            id={id}
+            className="dash-dropdown"
+            style={style}
+            data-dash-is-loading={
+                (loading_state && loading_state.is_loading) || undefined
+            }
+        >
+            <ReactDropdown
+                filterOptions={filterOptions}
+                options={sanitizeOptions(options)}
+                value={value}
+                onChange={onChange}
+                onInputChange={onInputChange}
+                backspaceRemoves={clearable}
+                deleteRemoves={clearable}
+                inputProps={{autoComplete: 'off'}}
+                {...omit(['setProps', 'value', 'options'], props)}
```

Comment:
Thanks for converting this component to functional - and of course getting rid of one `UNSAFE_` method 🎉  That will become very important as soon as someone tries to use Dash with React 18 (which is officially out as of a few weeks ago)

FWIW not necessary to include in this PR, but just because I see it carried over in your refactoring, I'd like us to eventually get rid of all the `ramda.omit` calls fowarding props to underlying components https://github.com/plotly/dash-core-components/issues/703